### PR TITLE
feat: Add netowrking configuration

### PR DESCRIPTION
**Pull request by alessio-creo**

### CODE REVIEW

File: src/main/21-networking.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.19.0"
+
+  name = "${var.app_name}-${var.env}-vpc"
+  cidr = "10.0.0.0/16"
+
+  azs             = ["${var.aws_region}a", "${var.aws_region}b", "${var.aws_region}c"]
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24","10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24","10.0.7.0/24", "10.0.8.0/24", "10.0.9.0/24"]
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/21-networking.tf]: Defined variables for all subnets configuration

File: src/main/21-networking.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.19.0"
+
+  name = "${var.app_name}-${var.env}-vpc"
```
[the remark this answers is not in the document]
Comment (alessio-creo) [reply src/main/21-networking.tf]: Name changed

File: src/main/98-variables.tf [hunk 3]
```diff
@@ -14,6 +14,26 @@ variable "env" {
   description = "Environment name"
 }
 
+variable "database_subnets" {
+  type        = list(string)
+  description = "Database dedicated private subnets"
+}
+
+variable "eks_control_plane_subnets" {
```
Comment (micdes-pagopa) [inline src/main/98-variables.tf:22]: Do we actually need to provision control plane subnets? EKS manages the control plane in its own "hidden" VPC

Comment (alessio-creo) [reply src/main/98-variables.tf:22]: Is right that control plane is in a AWS-managed VPC but eks_control_plane_subnets variable controls the subnets where the ENIs for the communication with the control plane are provisioned. Could be useful if should be increased the number of workload subnets without affecting the ENIs subnets.

Comment (micdes-pagopa) [reply src/main/98-variables.tf:22]: Ok now I get it, sorry I didn't understand the naming of the variable 😄

**Review by micdes-pagopa — APPROVED** [review]

**Pull request merged**